### DISCUSSION TITLE
The problem of deleting all files because of the icons file

### DISCUSSION THREAD

**Discussion by avcr2019** [airsdk/Adobe-Runtime-Support#2433] [General]
I would like to make icons for my application on the Apple system, and I used this file that I will attach to make icons, and indeed, icons have been made. But when I used it again, it erased all the files of the device completely, so how can I restore my project and my files, please
[assets-car-build.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/10435105/assets-car-build.zip)

**Comment by avcr2019** [comment]
please help me i lost all my file in my pc

**Comment by ajwfrost** [comment]
That script is probably not the safest thing to run without some extra checks on the variables...
```
DIR=`dirname $0`
rm -Rf $DIR/build
```
but that means it should have removed anything under `/build`, or hopefully under the build folder in whatever folder you're running the script from...

It shouldn't be possible to remove all of the files on your device (not sure whether you mean the files in the whole machine, or whether you were using the "build" location for other things?)

But in terms of restoring stuff, please take a look through the below suggestions, hopefully they still are in the Trash:
https://www.cleverfiles.com/howto/recover-deleted-files-using-mac-terminal.html

**Reply by avcr2019** [reply]
> https://www.cleverfiles.com/howto/recover-deleted-files-using-mac-terminal.html

Thank you for your reply
I do not know, because I tried the file and ran it for the first time yesterday, and there was no problem, and today I tried it in order to put the special icons, and I ran the file, and all the files that were on the device disappeared completely, and I went to my timeouts, there is nothing. And go to the desktop, there is nothing final. Everything that is not on the device is considered empty as its mother gave birth to. I don't know how, why and how this happened. I am very sad because the device had all my project files on it and so far I am afraid to restart the device so it does not open again and I am very afraid

**Reply by ajwfrost** [reply]
When you say "the device", what is it that you're talking about? Is it a macOS computer? in which case, on the bottom right of the dock, you should be able to see a 'bin' icon, if you right-click this and select "Open", it should list things that have been deleted. If you then right-click on something, you should see "Put Back" as an option.

Just be careful when clicking as all these menu items are next to ones that you really don't want to press...

**Reply by avcr2019** [reply]
> When you say "the device", what is it that you're talking about? Is it a macOS computer? in which case, on the bottom right of the dock, you should be able to see a 'bin' icon, if you right-click this and select "Open", it should list things that have been deleted. If you then right-click on something, you should see "Put Back" as an option.
> 
> Just be careful when clicking as all these menu items are next to ones that you really don't want to press..

Thank you again for the reply
My dear friend, my device is running Apple.
There is nothing in the trash for me to retrieve. There is nothing on the device at all, knowing the actual space for the files that were there, this space still exists, but the files where I do not know. I used a code to show hidden files, but without Joy, nothing appears on the device
